packages/core/src/module-resolver.ts

```diff
@@ -1494,16 +1509,10 @@ export class Resolver {
     if (engineConfig) {
       // we're directly inside an engine, so we're potentially resolvable as a
       // global component
-
-      // this kind of mapping is not true in general for all packages, but it
-      // *is* true for all classical engines (which includes apps) since they
-      // don't support package.json `exports`. As for a future v2 engine or app:
-      // this whole method is only relevant for implementing packageRules, which
-      // should only be for classic stuff. v2 packages should do the right
-      // things from the beginning and not need packageRules about themselves.
```

Comment:
This comment was claiming we don't care about making this code work in native v2 packages, but it honestly seems easier to just make it work than guarantee that nobody will try to ever use `reverseComponentLookup` in a v2 package.